File: scala/src/main/scala/tddmicroexercises/tirepressuremonitoringsystem/Alarm.scala
package tddmicroexercises.tirepressuremonitoringsystem

class Alarm
{
    val  LowPressureThreshold : Double = 17
    val HighPressureThreshold : Double = 21

    val sensor = new Sensor()

    var alarmOn = false

    def check()
    {
        val psiPressureValue = sensor.popNextPressurePsiValue()

        if (psiPressureValue < LowPressureThreshold || HighPressureThreshold < psiPressureValue)
        {
            alarmOn = true
        }
    }

    def isAlarmOn() : Boolean = {
        return alarmOn
    }
}

File: Swift/TirePressureMonitoringSystem/TirePressureMonitoringSystem/TirePressureMonitoringSystemTests/Alarm.swift
class Alarm {

    private let lowPressureThreshold = 17.0
    private let highPressureThreshold = 21.0

    let sensor = Sensor()

    var alarmOn = false

    func check() {
        let psiPressureValue = sensor.popNextPressurePsiValue()

        if (psiPressureValue < lowPressureThreshold || highPressureThreshold < psiPressureValue) {
            alarmOn = true
        }
    }

    func isAlarmOn() -> Bool {
        return alarmOn
    }
}
